sk-csharp-chatgpt-plugin/semantic-functions-generator/SemanticSkillGenerator.cs

```diff
@@ -0,0 +1,177 @@
+            {
+                continue;
+            }
+
+            string classSource = GenerateClassSource(rootNamespace!, folderName, folderGroup);
+            context.AddSource(folderName, SourceText.From(classSource, Encoding.UTF8));
+        }
+    }
+
+    // Generate the source code for a folder of semantic functions
+    private static string GenerateClassSource(string rootNamespace, string folderName, IGrouping<string, IGrouping<string, AdditionalText>> folderGroup)
+    {
+        // Use a StringBuilder to build the class source
+        StringBuilder functionsCode = new();
+
+        foreach (var functionGroup in folderGroup)
+        {
+            // Get the "skprompt.txt" and "config.json" files for this function
+            AdditionalText? configFile = functionGroup.FirstOrDefault(f => Path.GetFileName(f.Path).Equals(FunctionConfigFilename, StringComparison.InvariantCultureIgnoreCase));
+            AdditionalText? promptFile = functionGroup.FirstOrDefault(f => Path.GetFileName(f.Path).Equals(FunctionPromptFilename, StringComparison.InvariantCultureIgnoreCase));
+            if (promptFile != default && configFile != default)
+            {
+                functionsCode.AppendLine(GenerateFunctionSource(promptFile, configFile) ?? string.Empty);
+            }
+        }
+
+        return $@"/* ### GENERATED CODE - Do not modify. Edits will be lost on build. ### */
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+using AIPlugins.AzureFunctions.Extensions;
+
+namespace {rootNamespace};
+
+public class {folderName}
+{{
+    private readonly ILogger _logger;
+    private readonly IAIPluginRunner _pluginRunner;
+
+    public {folderName}(IAIPluginRunner pluginRunner, ILoggerFactory loggerFactory)
+    {{
+        this._pluginRunner = pluginRunner;
+        this._logger = loggerFactory.CreateLogger<{folderName}>();
+    }}
+
+    {functionsCode}
+}}";
+    }
+
+    private static string? GenerateFunctionSource(AdditionalText promptFile, AdditionalText configFile)
+    {
+        // Get the function name from the directory name
+        string? functionName = Path.GetFileName(Path.GetDirectoryName(promptFile.Path));
+        if (string.IsNullOrWhiteSpace(functionName)) { return null; }
+
+        string? metadataJson = configFile.GetText()?.ToString();
+        if (string.IsNullOrWhiteSpace(metadataJson)) { return null; }
+
+        // Get the function description from the config file
+        PromptConfig? config = JsonConvert.DeserializeObject<PromptConfig>(metadataJson!);
+        if (config == null) { return null; }
+
+        string descriptionProperty = string.IsNullOrWhiteSpace(config.Description)
+            ? string.Empty
+            : $@", Description = ""{config.Description}""";
```

Comment:
Are we OK with empty description? I'd expect the planner won't be able to use the plugin.